components/Themes/index.js

```diff
@@ -40,86 +40,107 @@ const ThemeItem = ({ children, item, isSelected, onClick }) => (
 
 const themeIcon = <ThemeIcon />
 
+const getCustomName = themes =>
+  `Custom Theme ${themes.filter(({ name }) => name.startsWith('Custom Theme')).length + 1}`
+
 class Themes extends React.PureComponent {
   selectedTheme = DEFAULT_THEME
 
   state = {
     themes: THEMES,
     preset: this.props.theme,
-    highlights: {},
-    name: 'Custom Theme',
+    input: 'Custom Theme',
     selected: null
   }
 
   dropdown = React.createRef()
 
   componentDidMount() {
+    const { update, theme, highlights } = this.props
     const storedThemes = getThemes(localStorage) || []
 
     this.setState(({ themes }) => {
       const newThemes = [...storedThemes, ...themes]
 
-      const name = `Custom Theme ${newThemes.filter(({ name }) => name.startsWith('Custom Theme'))
-        .length + 1}`
+      this.selectedTheme = newThemes.find(({ id }) => id === theme) || DEFAULT_THEME
```

Comment:
Should this line and the following `update` be moved into the callback in the second `setState` argument?